### PR TITLE
Execution role

### PR DESCRIPTION
**Pull request by hsl89**
*Issue #, if available:*

*Description of changes:*
Notebook 1 of the sagemaker-fundamentals series;
* describes the basics of IAM role and polices
* creates roles and policies via boto3

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by paul-cb — APPROVED** [review]
Runs. Looks good, approved.

File: sagemaker-fundamentals/execution-role/execution-role.ipynb [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+    "time.sleep(15)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test your role\n",
+    "\n",
+    "Now, you can verify that the role we just created (`basic-role`) is allowed to list all S3 buckets under your account and it is not allowed to do anything else with your AWS resource. "
```
Comment (ngluna) [inline sagemaker-fundamentals/execution-role/execution-role.ipynb:250]: Re-word: "You can now verify..."

"....anything else with your Amazon Resource Name (ARN)."

File: sagemaker-fundamentals/execution-role/execution-role.ipynb [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+    "pp.pprint(res)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Amazon Managed Policies\n",
+    "\n",
+    "So far you have learnt how to create an IAM role and how to grant it permissions on your AWS resources via IAM polices. The example you went through above uses S3 buckets as an example service, but the same idea can be generalized when you need to define more complicated policies on more services. \n",
```
Comment (ngluna) [inline sagemaker-fundamentals/execution-role/execution-role.ipynb:433]: "learnt" -> "learned"

**Review by ngluna — COMMENTED** [review]
Nice notebook. Thanks for following Python PEP 257 by including doc strings in your functins.

File: sagemaker-fundamentals/execution-role/execution-role.ipynb [hunk 3]
```diff
@@ -0,0 +1,582 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# IAM Role\n",
+    "\n",
+    "In this notebook, we review the concept of IAM role and set up one to be used for later notebooks in `/sagemaker-fundamentals`. An IAM role is an identity associated with your AWS account that has pre-configured permission policies that determine what this role can do and with respect to your AWS resources. For example, you can define a role that can do *everything* to your S3 resource but nothing else. Such an role is useful if it is *assumed* by a program that synchronize your data from your local machine to an S3 bucket. You can be certain that this program would not accidently create EC2 instances that incur higher costs. \n",
```
[the remark this answers is not in the document]
Comment (hsl89) [reply sagemaker-fundamentals/execution-role/execution-role.ipynb:9]: you mean I should change "... we review the concept of IAM role ..." to "... we review the concept of IAM Roles ..."?

File: sagemaker-fundamentals/execution-role/execution-role.ipynb (hunk 2 above)
Comment (hsl89) [reply sagemaker-fundamentals/execution-role/execution-role.ipynb:433]: learned

**Comment by hsl89** [comment]
hey @ngluna, since I staged (accidently) execution-role.ipynb in create-trainining-job branch, I incopred your suggested there https://github.com/aws/amazon-sagemaker-examples/pull/2050

**Pull request merged**